transcom/mymove
# B-21094: Automatic Reweigh Based on Estimated Weight

## Agility Ticket: [B-21094](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21094)

## Agility Ticket's Acceptance Criteria

As the MilMove system (on behalf of the TOO),
I need to request a reweigh for all shipments on a customer's move based on the sum of the estimated weight(s),
So that I can ensure the customer has the greatest chance of reweighs being performed on all of their shipments (through the use of more advanced notice).

AC1: 
Given the sum of all estimated weights on a customer's move,
When that sum is within 10% (or greater) of the customer's weight entitlement,
Then a reweigh is requested for all shipments on a that move (excludes PPMs).

AC2: PRESERVE EXISTING CAPABILITY ALREADY IN MILMOVE
Given a customer's move where the sum of the estimated weights did not result in automatic reweigh(s) being requested,
When the sum of the actual shipment weights is within 10% (or greater) of the customer's weight entitlement,
Then a reweigh is requested for all shipments on that move (excludes PPMs).

## How to test

1. As a **_Service Member_**, create a new move + orders + at least 2 non-PPM shipments.
2. As **_Service Counsellor_**, update the move's orders and approve/send the move.
3. As **_TOO_**, approve the shipments (this should make the move available to the Prime user role).
4. As **_Prime_**, update each shipment so that the sum of the Estimated Weights of all the shipments on the move falls within the top 10% of the Authorized Weight.
___
Note 1: _Currently you must enter both Estimated and Actual Weights on each of the shipments if you are using the Prime Simulator because current form validation forces you to do so.. however, this is a ticket which should change this functionality so that the Prime Simulator can have Estimated Weight without the need for Actual Weight (and vice versa)._
___
Note 2: _The reason the sum needs to fall within the top 10% of the Authorized Weight is because other logic may be carried out once the sum of Estimated Weights or Actual Weights goes over the Authorized Weight._
___
```
Authorized Weight = A
Target Weight Floor = T
A * 0.9 = T
```
Example: If Authorized Weight is 7000 lbs, 7000 * 0.9 = 6300. This means that, from the entire move, if the sum of the Estimated Weights is at least 6300 lbs or the sum of the Actual Weights is at least 6300 lbs, it will trigger an automatic reweight request.

6. As soon as the sum of the Estimated Weights of the shipments on the move falls within the top 10% of the Authorized Weight, you should see the "Reweigh Weight" attribute (seen below) with the red "Missing" text for each shipment on the move. This indicates that an auto-reweigh has been initiated.
<img width="443" alt="Screenshot 2024-12-05 at 6 32 37 AM" src="https://github.com/user-attachments/assets/067dec9d-eba6-4aa0-95e6-a26bdba22ce8">
